src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1175 @@ const ActorInit En_Kakasi_InitVars = {
+        Math_ApproachF(&this->unk220.y, this->unk244.y, 0.4f, 4.0f);
+        Math_ApproachF(&this->unk220.z, this->unk244.z, 0.4f, 4.0f);
+
+        Math_ApproachF(&this->unk20C, this->unk210, 0.3f, 10.0f);
+
+        func_8016970C(globalCtx, this->cutsceneCamId, &this->unk220, &this->unk214);
+        func_80169940(globalCtx, this->cutsceneCamId, this->unk20C);
+    }
+}
+
+/*
+ * goes off every frame of song teach, but... doing what?
+ */
+void func_8096FBB8(EnKakasi* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.unk12048 == 0 || globalCtx->msgCtx.unk12048 == 1 || globalCtx->msgCtx.unk12048 == 2 ||
+        globalCtx->msgCtx.unk12048 == 3 || globalCtx->msgCtx.unk12048 == 4) {
+        // why not 0 < x < 4? fewer branches
+        this->unk190++;
+    }
+    if (this->unk190 != 0 && this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+    }
+    if (this->unk190 > 8) {
+        this->unk190 = 8;
+    }
+    if (this->unk190 != 0) {
+        Math_ApproachF(&this->skelanime.animPlaybackSpeed, 1.0f, 0.1f, 0.2f);
+        this->actor.shape.rot.y = this->actor.shape.rot.y + 0x800;
+    }
+}
+
+void EnKakasi_InitTimeSkipDialogue(EnKakasi* this) {
+    EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SLOWROLL);
+    this->actionFunc = EnKakasi_TimeSkipDialogue;
+}
+
+void EnKakasi_TimeSkipDialogue(EnKakasi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (gSaveContext.respawnFlag != -4 && gSaveContext.respawnFlag != -8) {
+        if (gSaveContext.time != 0x4000 && gSaveContext.time != 0xC000 && (gSaveContext.eventInf[1] & 0x80) == 0) {
+            if (this->actor.textId == 0) {
+                // "Oh, yeah!  How was it? It went by in an instant, right?  I'm still full of energy!"
+                this->actor.textId = 0x1653;
+                gSaveContext.weekEventReg[0x53] &= 0xFE; // a lot of flags
+                this->unkMsgState1AC = 5;
+                player->stateFlags1 |= 0x20;
+                this->actor.flags |= 0x10000;
+            }
+            if (func_800B84D0(&this->actor, globalCtx) != 0) {
+                player->stateFlags1 &= ~0x20;
+                this->unkState196 = 2;
+                this->actor.flags &= ~0x10000;
+                this->actionFunc = EnKakasi_RegularDialogue;
+            } else {
+                func_800B8500(&this->actor, globalCtx, 9999.9f, 9999.9f, -1);
+            }
+        }
+    }
+}
+
+void EnKakasi_SetupIdleStanding(EnKakasi* this) {
+    this->unkState196 = 0;
+    this->actionFunc = EnKakasi_IdleStanding;
+}
+
+void EnKakasi_IdleStanding(EnKakasi* this, GlobalContext* globalCtx) {
+    u32 saveContextDay;
+    s16 passedValue1;
+    s16 passedValue2;
+
+    saveContextDay = gSaveContext.day;
+    // "Yo! Hey, baby!  I'm a stylin' scarecrow wandering in search of pleasant music.
+    // Time will pass in the blink of an eye if you dance with me."
```

Comment:
We avoid leaving text dump in the code of an actor.
You could write some vague text referring to what the actor says, but not the literal dialogue.

Reply:
Oh I kinda already did this for ginko, didnt think exact text would be an issue, I actually thought if anything paraphrasing would be the bigger issue as its more ambigious for documenting.

Guess I can go... abstract some text?